sw/device/lib/dif/dif_spi_device.h

```diff
@@ -0,0 +1,243 @@
+ */
+dif_spi_device_result_t dif_spi_device_irq_get(const dif_spi_device_t *spi,
+                                               dif_spi_device_irq_type_t type,
+                                               bool *flag_out);
+
+/**
+ * Clears all active interrupt bits.
+ *
+ * @param spi A SPI device.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_clear_all(
+    const dif_spi_device_t *spi);
+
+/**
+ * Enable a particular interrupt enable bit to enabled or disabled.
+ *
+ * @param spi A SPI device.
+ * @param type Which IRQ type to toggle.
+ * @param state The state to update the bit to.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_enable(
+    const dif_spi_device_t *spi, dif_spi_device_irq_type_t type,
+    dif_spi_device_irq_state_t state);
+
+/**
+ * Forces a particular IRQ type to fire.
+ *
+ * @param spi A SPI device.
+ * @param type Which IRQ type to fire.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_irq_force(
+    const dif_spi_device_t *spi, dif_spi_device_irq_type_t type);
+
+/**
+ * Sets up the "FIFO level" (that is, number of bytes present in a particular
+ * FIFO) at which the TxLevel and RxLevel IRQs will fire.
+ *
+ * For the reciept side, when the FIFO overflows `rx_level`, an interrupt is
+ * fired. This can be used to detect that more data should be read from the
+ * RX FIFO. This is the `SRAM -> Main Memory` case.
+ *
+ * Conversely, for the transmission side, when the FIFO underflows `tx_level`,
+ * an interrupt is fired. This can be used to detect that there is free space to
+ * write more data to the TX FIFO. This is the `Main Memory -> SRAM` case.
+ *
+ * @param spi A SPI device.
+ * @param rx_level The new RX level, as described above.
+ * @param tx_level The new TX level, as described above.
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_set_irq_levels(
+    const dif_spi_device_t *spi, uint16_t rx_level, uint16_t tx_level);
+
+/**
+ * Checks how many bytes are still pending reciept by software in the RX FIFO.
+ *
+ * @param spi A SPI device.
+ * @param bytes_pending Out param for the number of bytes pending
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_rx_pending(const dif_spi_device_t *spi,
+                                                    size_t *bytes_pending);
+
+/**
+ * Checks how many bytes are still pending transmission by hardware in the TX
+ * FIFO.
+ *
+ * @param spi A SPI device.
+ * @param bytes_pending Out param for the number of bytes pending
+ * @return The result of the operation.
+ */
+dif_spi_device_result_t dif_spi_device_tx_pending(const dif_spi_device_t *spi,
+                                                    size_t *bytes_pending);
+
+/**
+ * Reads at most `buf_len` bytes from the RX FIFO; the number of bytes read
+ * will be written to `bytes_recved`.
```

Comment:
`bytes_recvd` or `bytes_received`. `bytes_receved` is weird. 😛

Reply:
Fiiine.

Reply:
No, I missed updating the comment after I updated the variable name. Whoops.